src/pyodide/types/emscripten.d.ts

```diff
@@ -8,22 +8,34 @@ interface PyodideConfig {
   jsglobals: any;
   resolveLockFilePromise?: (lockfile: PackageLock) => void;
   indexURL?: string;
+  _makeSnapshot?: boolean;
 }
 
+type SerializedHiwireValue = { path: string[] } | { serialized: any } | null;
+
+type SnapshotConfig = {
+  hiwireKeys: SerializedHiwireValue[];
+  immortalKeys: string[];
+};
+
 interface API {
   config: PyodideConfig;
-  finalizeBootstrap: () => void;
+  finalizeBootstrap: (
+    fromSnapshot?: SnapshotConfig,
+    snapshotDeserializer?: (obj: any) => any
+  ) => void;
   public_api: Pyodide;
   rawRun: (code: string) => [status: number, err: string];
   initializeStreams: (
     stdin?: any,
     stdout?: (a: string) => void,
     stderr?: (a: string) => void
   ) => void;
-  version: string;
+  version: '0.26.0a2' | '0.27.5';
```

Comment:
What happens when we add new versions? It may be difficult to remember to update this, maybe best to keep it as a string and add some `throw` statements when the version isn't what we expect

Reply:
This will make the editor autocomplete the version everywhere which is really handy. Typescript will remind us to update this when we need to.